### PR TITLE
Refactor package selection and installation

### PR DESCRIPTION
**Pull request by rsalevsky**
- install cloud packages only when the cloud repo is available
- install SES only on SLE12

Superseeds #422

### CODE REVIEW

**Comment by tboerger** [comment]
I'm missing an explanation why this change is needed

File: chef/cookbooks/provisioner/recipes/update_nodes.rb [hunk 1]
```diff
@@ -225,14 +225,32 @@
...
               append << "UseSSH=1 SSHPassword=#{ssh_password}" if ssh_password
             end
-          end
 
-          packages = node[:provisioner][:packages][os] || []
+            if node[:provisioner][:suse][:cloud_available]
+              packages.push("supportutils-plugin-susecloud")
+
+              if target_platform_version == "11.3"
+                packages.push("use-sle11-openstack-cloud-release")
```
Comment (vuntz) [inline chef/cookbooks/provisioner/recipes/update_nodes.rb]: Typo: use => suse

**Comment by vuntz** [comment]
I must admit I'm not sure I understand the rationale for moving this inside the recipe. What was the issue with having it in the template?

**Comment by rsalevsky** [comment]
Since #420 we give anyway a list of packages to the template. My basic idea of this PR is to get rid of the is_ses attribute because the code complexity is too height. And adding the additional check from #422 makes the situation not better. 
While this I moved also the other packages because having the packages at one place makes it easier to maintain the code. Otherwise you have to look at two points for knowing which packages will be installed.

File: chef/cookbooks/provisioner/recipes/update_nodes.rb [hunk 2]
```diff
@@ -225,14 +225,32 @@
...
+              else
+                packages.push("suse-openstack-cloud-release")
+              end
+            end
+
+            if node[:provisioner][:suse][:storage_available]
```
Comment (vuntz) [inline chef/cookbooks/provisioner/recipes/update_nodes.rb]: So I don't think the logic is correct. If Cloud and Storage are available, we don't want to install ses-release automatically.

**Comment by vuntz** [comment]
@rsalevsky to be honest, I'm not sold on the idea that this should be moved from the template to the recipe. And I'm still a bit confused as to what is wrong with the original code :-)

**Comment by rsalevsky** [comment]
@vuntz The code itself is fine but I think having the if else stuff in the template is not correct. In my opinion a template should not contain to much code especially not when this can be done easily in the script itself. It makes the template unclean and it's hard to see whats going on.

So it's just my personal opinion so feel free to close the PR when I'm wrong.

**Comment by vuntz** [comment]
> So it's just my personal opinion so feel free to close the PR when I'm wrong.

I don't think you're "wrong", I think it's a matter of taste. So let's see what others think before deciding which way to go.

File: chef/cookbooks/provisioner/recipes/update_nodes.rb (hunk 2 above)
Comment (tboerger) [inline chef/cookbooks/provisioner/recipes/update_nodes.rb]: This should be only a check for is_ses

**Comment by tboerger** [comment]
I like to keep templates simple, but maybe it's already enough to put the packages with a loop into the template.

**Pull request closed without merging**